packages/contracts/contracts/minter-suite/Minters/MinterSEAV1.sol

```diff
@@ -0,0 +1,1169 @@
+     * rather the timestamp after which a auction may be started. Also note
+     * that the passed value here must either be in the future, or `0` (which
+     * indicates that auctions are immediately startable).
+     * @param _auctionDurationSeconds Duration of new auctions, in seconds,
+     * before any extensions due to bids being placed inside buffer period near
+     * the end of an auction.
+     * @param _basePrice reserve price (minimum starting bid price), in wei.
+     * Must be greater than 0, but may be as low as 1 wei.
+     * @param _minBidIncrementPercentage Minimum percentage amount that a new
+     * bid must be over the previous bid. Must be greater than 0, and may be as
+     * high as 255. Recommended value is 5%. Values represent whole
+     * percentages, so a value of 5 represents 5%.
+     * @dev `_basePrice` of zero not allowed to prevent accidental
+     * misconfiguration of auctions
+     */
+    function configureFutureAuctions(
+        uint256 _projectId,
+        address _coreContract,
+        uint256 _timestampStart,
+        uint256 _auctionDurationSeconds,
+        uint256 _basePrice,
+        uint8 _minBidIncrementPercentage
+    ) external {
+        // CHECKS
+        _onlyNonZero(_basePrice);
+        _onlyNonZero(_minBidIncrementPercentage);
+        AuthLib.onlyArtist({
+            _projectId: _projectId,
+            _coreContract: _coreContract,
+            _sender: msg.sender
+        });
+        SEALib.SEAProjectConfig storage _SEAProjectConfig = _SEAProjectConfigs[
+            _coreContract
+        ][_projectId];
+        MaxInvocationsLib.MaxInvocationsProjectConfig
+            storage _maxInvocationsProjectConfig = _maxInvocationsProjectConfigMapping[
+                _coreContract
+            ][_projectId];
+        require(
+            _timestampStart == 0 || block.timestamp < _timestampStart,
+            "Only future start times or 0"
+        );
+        require(
+            _auctionDurationSeconds >= MIN_AUCTION_DURATION_SECONDS,
+            "Auction duration below minimum"
+        );
+        // EFFECTS
+        _SEAProjectConfig.timestampStart = _timestampStart.toUint64();
+        _SEAProjectConfig.auctionDurationSeconds = _auctionDurationSeconds
+            .toUint32();
+        _SEAProjectConfig.basePrice = _basePrice;
+        _SEAProjectConfig
+            .minBidIncrementPercentage = _minBidIncrementPercentage;
+
+        emit ConfiguredFutureAuctions({
+            projectId: _projectId,
+            coreContract: _coreContract,
+            timestampStart: _timestampStart.toUint64(),
+            auctionDurationSeconds: _auctionDurationSeconds.toUint32(),
+            basePrice: _basePrice,
+            minBidIncrementPercentage: _minBidIncrementPercentage
+        });
+
+        // sync local max invocations if not initially populated
+        if (
+            !MaxInvocationsLib.maxInvocationsIsInitialized(
+                _maxInvocationsProjectConfig
```

Comment:
@jakerockland - do you like the use of this library function here? If so, I can copy it to the other minters as well. If not, we can remove the function from the library, and make this minter consistent with the other minters.

Reply:
I do dig this yes – I think that is a great suggestion.

One nit, would suggest that we make the conditional branch here start with the positive though, in the way of:

```suggestion
        // sync local max invocations if not initially populated
        if (
            MaxInvocationsLib.maxInvocationsIsInitialized(
                _maxInvocationsProjectConfig
            )
        ) {
            // for convenience, try to mint to next token slot
            _tryMintTokenToNextSlot(_projectId, _coreContract);
        } else {
            syncProjectMaxInvocationsToCore(_projectId, _coreContract);
            // @dev syncProjectMaxInvocationsToCore function calls
            // _tryMintTokenToNextSlot, so we do not call it here.
        }
```

Reply:
good call - updated in [c75484d](https://github.com/ArtBlocks/artblocks-contracts/pull/871/commits/c75484d7248638a68120c967015d5211419802b2)